src/util/copy/mod.rs

```diff
@@ -107,18 +107,8 @@ impl<VM: VMBinding> GCWorkerCopyContext<VM> {
     /// * `bytes`: The size of the object in bytes.
     /// * `semantics`: The copy semantic used for the copying.
     pub fn post_copy(&mut self, object: ObjectReference, bytes: usize, semantics: CopySemantics) {
-        // Clear forwarding bits if the forwarding bits are in the header.
-        if !VM::VMObjectModel::LOCAL_FORWARDING_BITS_SPEC.is_on_side() {
-            object_forwarding::clear_forwarding_bits::<VM>(object);
-        } else {
-            // NOTE: If the forwarding bits were on the side, they would have been cleared when
-            // preparing the space.  See:
-            // - `CopySpace::prepare` and
-            // - `PrepareBlockState::reset_object_mark`.
-            debug_assert!(!object_forwarding::is_forwarded_or_being_forwarded::<VM>(
-                object
-            ));
-        }
+        // Clear forwarding bits.
+        object_forwarding::clear_forwarding_bits::<VM>(object);
```

Comment:
This reverts the changes in https://github.com/mmtk/mmtk-core/commit/d341506b70a4465dbbc05e87d27d0b7bd23844bb. It is okay.

Reply:
Yes.  The change in https://github.com/mmtk/mmtk-core/commit/d341506b70a4465dbbc05e87d27d0b7bd23844bb was an optimisation, but was wrong.  I didn't take free chunks into consideration.